File: orchestrator-java-reactive/src/main/java/com/github/mzr/coder/orchestrator/java/reactive/service/RiskService.java
package com.github.mzr.coder.orchestrator.java.reactive.service;

import com.github.mzr.coder.model.ServiceRequest;
import com.github.mzr.coder.model.ServiceResponse;
import reactor.core.publisher.Mono;

public interface RiskService {
    String getName();
    Mono<ServiceResponse> execute(ServiceRequest request);
}

File: orchestrator-kotlin-coroutine/src/main/java/com/github/mzr/coder/orchestrator/kotlin/reactive/service/RiskService.kt
package com.github.mzr.coder.orchestrator.kotlin.reactive.service

import com.github.mzr.coder.model.ServiceRequest
import com.github.mzr.coder.model.ServiceResponse
import reactor.core.publisher.Mono

interface RiskService {
    val name: String
    suspend fun execute(request: ServiceRequest): ServiceResponse
}